spec/consumer_spec.rb

```diff
@@ -0,0 +1,65 @@
+require_relative 'spec_helper'
+
+describe Consumer do
+
+  let(:config) {{
```

Comment:
This is also in the content-register ([message_queue_consumer_spec.rb](https://github.com/alphagov/content-register/blob/a1948fe2f4c655630880042b8396b8fe4cae6f6d/spec/lib/message_queue_consumer_spec.rb)), but do you think it's worth it to update the spec style here?

Getting rid of some of the `let`s here might make it more readable (https://robots.thoughtbot.com/lets-not) and using `expect(x).to have_received(y)` makes the tests read in the order they were intended (https://robots.thoughtbot.com/four-phase-test).

for example:

```ruby
it "subscribes to a queue" do
  expect(queue).to receive(:bind)
  expect(queue).to receive(:subscribe)

  Consumer.new(config, client_processor).run
end
```

could be something like

```ruby
it "subscribes to a queue" do
  queue = build_queue
  consumer = build_consumer_with_default_config(queue)
  
  consumer.run
 
  expect(queue).to have_received(:bind)
  expect(queue).to have_received(:subscribe)
end
```

It's more verbose, but it reads like you want it to: "I have a queue, I put that queue in my consumer, and after the consumer runs I expect these methods to have been called."

(I have no problem with the code as-is, but I thought this would be a nice place to start evangelising that thoughtbot testing style as I really like it).